.github/workflows/protocol_tests.yml

```diff
@@ -38,6 +38,30 @@ jobs:
       - name: Compile Contracts
         run: forge compile
 
-      - name: Run tests
+      - name: Run tests common
         # can't use gas limit because some setUp function use more than the limit
-        run: forge test -vvv # --block-gas-limit 20000000
\ No newline at end of file
+        run: forge test -vvv --match-path "test-sol/common/*" # --block-gas-limit 20000000 
+
+      - name: Run tests compatibility
+        if: success() || failure()
+        run: forge test -vvv --block-gas-limit 20000000 --match-path "test-sol/compatibility/*"
+
+      - name: Run tests governance/network
+        if: success() || failure()
+        run: forge test -vvv --block-gas-limit 20000000 --match-path "test-sol/governance/network/*"
+
+      - name: Run tests governance/validators
+        if: success() || failure()
+        run: forge test -vvv --block-gas-limit 20000000 --match-path "test-sol/governance/validators/*"
+      
+      - name: Run tests governance/voting
+        if: success() || failure()
+        run: forge test -vvv --block-gas-limit 20000000 --match-path "test-sol/governance/voting/*"
+
+      - name: Run tests stability
+        if: success() || failure()
+        run: forge test -vvv --block-gas-limit 20000000 --match-path "test-sol/stability/*"
+
+      - name: Fail if there are tests without folder
```

Comment:
We could use github actions `matrix` for this and then check if any of the tests in any of the folders (or root) were not ran at all

Reply:
That's an option, although I believe `matrix` works on jobs, not inside a job. Some sort of looping and automatic detection of directories would be great though, especially when new ones are added, and it would render the last step redundant.